app/knights/knight.py

```diff
@@ -0,0 +1,24 @@
+class Knight:
+
+    def __init__(self, config: dict) -> None:
+
+        self.name = config["name"]
+        self.power = config["power"]
+        self.hp = config["hp"]
+        self.protection = 0
+
+        if len(config["armour"]) > 0:
+            for item in config["armour"]:
+                self.protection += item["protection"]
+
+        self.power += config["weapon"]["power"]

```

Comment:
Ensure that 'weapon' and 'power' keys exist in the config dictionary to avoid KeyError. Consider using the `get` method with default values or adding error handling.